### PR TITLE
Focused Launch: Implement plan step in summary

### PR DESCRIPTION
**Pull request by alshakero**
### Changes proposed in this Pull Request

* This implements the plan step in summary page. 

### What's left
- [x] Figure out the plan's name when the user has a paid plan
- [x] Figure out an i18n way to show plan prices. 

### Testing instructions

1. In the project root, run `yarn && yarn start`. In a separate terminal window, run `cd apps/editing-toolkit && yarn dev --sync`.
2. Sandbox testingfocusedflow.wordpress.com.
3. Sandbox newffv5t45654654645.wordpress.com.
4. In your sandbox, checkout D52710-code.

#### Without existing premium plan
5. Visit https://testingfocusedflow.wordpress.com/wp-admin/post-new.php.
6. In console, type `wp.data.dispatch('automattic/launch' ).openFocusedLaunch()`.
7. You should see Domain and Plan steps active. 
8. Picking a paid domain should unselect and lock the free plan.

#### With an existing premium plan
5. Visit https://newffv5t45654654645.wordpress.com/wp-admin/post-new.php.
6. In console, type `wp.data.dispatch('automattic/launch' ).openFocusedLaunch()`.
7. You should see the Domain step locked with a premium plan. 
8. You should be able to switch between domains.

Fixes part of #46865 and #47242

### CODE REVIEW

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 1]
```diff
@@ -296,6 +311,65 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
...
+									/>
+								</FocusedLaunchSummaryItem>
+							) : (
+								[
+									<FocusedLaunchSummaryItem
+										readOnly={ hasPaidDomain || selectedPaidDomain }
+										onClick={ () => defaultFreePlan && onSetPlan( defaultFreePlan ) }
+										isSelected={ ! ( hasPaidDomain || selectedPaidDomain ) && selectedPlan?.isFree }
+									>
+										<LeadingContentSide label={ __( 'Free plan', __i18n_text_domain__ ) } />
```
Comment (a8ci18n) [inline packages/launch/src/focused-launch/summary/index.tsx:334]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **24** times:
`translate( 'Free Plan', { context: 'Short plan name on storage button'} )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-step-focused-launch)

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 2]
```diff
@@ -199,7 +199,7 @@ const DomainStep: React.FunctionComponent< DomainStepProps > = ( {
 						<p className="focused-launch-summary__side-commentary-title">
 							{ createInterpolateElement(
 								__(
-									'<strong>Unique domains</strong> help build brand recongnition and trust',
+									'<strong>Unique domains</strong> help build brand recognition and trust',
```
Comment (a8ci18n) [inline packages/launch/src/focused-launch/summary/index.tsx:202]: :warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **1** times:
`translate( '<strong>Unique domains</strong> help build brand trust' )` *ES Score: 8*

**Review by ciampo — CHANGES_REQUESTED** [review]
The plan step becomes locked immediately after the Focused Launch modal loads

See recording https://d.pr/v/WzuJji

Maybe I'm not testing for those changes correctly?

File: packages/launch/src/focused-launch/summary/index.tsx (hunk 2 above)
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:202]: Thank you for fixing my typo!

Comment (alshakero) [reply packages/launch/src/focused-launch/summary/index.tsx:202]: Highly recommend the "Code Spell Checker" extension. Saves you from bugs too!

File: packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx [hunk 3]
```diff
@@ -30,19 +30,22 @@ export const LeadingContentSide: React.FunctionComponent< LeadingSideProps > = (
...
 
 export const TrailingContentSide: React.FunctionComponent< TrailingSideProps > = ( props ) => {
-	if ( props.price ) {
-		return <div className="focused-launch-summary-item__price-cost">{ props.price }</div>;
+	if ( props.warningNote ) {
+		return <div className="focused-launch-summary-item__warning-note">{ props.warningNote }</div>;
 	}
-	return <div className="focused-launch-summary-item__warning-note">{ props.warningNote }</div>;
+	return <div className="focused-launch-summary-item__price-cost">{ props.price }</div>;
 };
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx:37]: For more clarity and future-proofing, we should probably refactor `TrailingSideProps` to be something like

```
{
  node: ReactNode;
  noteType: 'PRICE' | 'WARNING';
}
```

and refactor the component's code to something like

```
return <div
  className={classNames({
    'focused-launch-summary-item__warning-note': noteType === 'WARNING',
    'focused-launch-summary-item__price-cost': noteType === 'PRICE',
  })}
>
  { props.node }
</div>;
```

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 4]
```diff
@@ -277,7 +306,19 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
...
 						</label>
-						{ /* @TODO: insert locked purchased plan item here */ }
+						<div>
+							<FocusedLaunchSummaryItem readOnly={ true }>
+								<LeadingContentSide label={ sitePlan?.product_name_short_with_suffix } />
+								<TrailingContentSide
+									price={
+										<>
+											<Icon icon={ check } size={ 18 } />{ ' ' }
+											{ __( 'Purchased', __i18n_text_domain__ ) }{ ' ' }
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:316]: Is the trailing `{ ' ' }` necessary?

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 5]
```diff
@@ -296,6 +337,47 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
...
+										defaultPaidPlan?.isPopular ? __( 'Popular', __i18n_text_domain__ ) : ''
+									}
+								/>
+								<TrailingContentSide
+									price={
+										<>
+											<span>{ defaultPaidPlan && planPrices[ defaultPaidPlan?.storeSlug ] }</span>
+											<span>
+												{
+													// translators: /mo is short for "per-month"
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:372]: Nice touch!

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 6]
```diff
@@ -296,6 +337,47 @@ const PlanStep: React.FunctionComponent< PlanStepProps > = ( {
...
+								<LeadingContentSide
+									label={ defaultPaidPlan?.titleWithPlanSuffix }
+									badgeText={
+										defaultPaidPlan?.isPopular ? __( 'Popular', __i18n_text_domain__ ) : ''
+									}
+								/>
+								<TrailingContentSide
+									price={
+										<>
+											<span>{ defaultPaidPlan && planPrices[ defaultPaidPlan?.storeSlug ] }</span>
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:369]: Not a blocker, but I think that we should refactor the code to handle better the lack of data (ie. when it's loading, or when it's failing).

I opened #47410 to keep track of the loading aspect

File: packages/launch/src/focused-launch/summary/index.tsx [hunk 7]
```diff
@@ -351,9 +433,14 @@ const Summary: React.FunctionComponent = () => {
...
 
 	const { sitePrimaryDomain, siteSubdomain, hasPaidDomain } = useSiteDomains();
+	const selectedPlan = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedPlan() );
 	const selectedDomain = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedDomain() );
-	const { setDomain, unsetDomain } = useDispatch( LAUNCH_STORE );
+	const { setDomain, unsetDomain, setPlan, unsetPlan } = useDispatch( LAUNCH_STORE );
 	const domainSearch = useDomainSearch();
+	const defaultPaidPlan = useSelect( ( select ) => select( PLANS_STORE ).getDefaultPaidPlan() );
+	const defaultFreePlan = useSelect( ( select ) => select( PLANS_STORE ).getDefaultFreePlan() );
+	const planPrices = useSelect( ( select ) => select( PLANS_STORE ).getPrices() );
```
Comment (ciampo) [inline packages/launch/src/focused-launch/summary/index.tsx:442]: Should these selectors be refactored into a `usePlans()` hook?

**Comment by ciampo** [comment]
> The plan step becomes locked immediately after the Focused Launch modal loads
> 
> See recording https://d.pr/v/WzuJji
> 
> Maybe I'm not testing for those changes correctly?

After some debugging, a non-blocking bug has been found. @alshakero opened issue #47412 to track it separately

File: packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx (hunk 3 above)
Comment (alshakero) [reply packages/launch/src/focused-launch/summary/focused-launch-summary-item/index.tsx:37]: Love this.

**Comment by alshakero** [comment]
Addressed all 🙏

**Review by ciampo — APPROVED** [review]
Changes LGTM

**Pull request merged**